rpcs3/rpcs3qt/trophy_manager_dialog.cpp

```diff
@@ -64,7 +64,7 @@ trophy_manager_dialog::trophy_manager_dialog() : QWidget(), m_sort_column(0), m_
 
 	// Populate the trophy database
 	QDirIterator dir_iter(qstr(vfs::get(m_TROPHY_DIR)));
-	while (dir_iter.hasNext()) 
+	while (true) 
```

Comment:
You are fixing the wrong thing. This is java-style iterator.

So after construction it point before first entry. So the `while (dir_iter.hasNext()) ` was correct.

The `if (dir_iter.fileName() == "")` seems wrong.

I think `dir_iter.next();` should've been called at the top. I.e. something like (not tested):

```cpp
	while (dir_iter.hasNext()) 
	{
		dir_iter.next();
		if (dir_iter.fileName() == "" || dir_iter.fileName() == "." || dir_iter.fileName() == ".." || dir_iter.fileName() == ".gitignore")
		{
			continue;
		}
		std::string dirName = sstr(dir_iter.fileName());
		LOG_TRACE(GENERAL, "Loading trophy dir: %s", dirName);
		LoadTrophyFolderToDB(dirName);
	}
```

Reply:
Yeah that's right.